scripts/macros/spells/spiritOfDeath.js

```diff
@@ -0,0 +1,221 @@
+            },
+            prototypeToken: {
+                name,
+                disposition: workflow.token.document.disposition
+            },
+            items: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData]
+        },
+        token: {
+            name,
+            disposition: workflow.token.document.disposition
+        }
+    };
+    let avatarImg = itemUtils.getConfig(workflow.item, 'avatar');
+    let tokenImg = itemUtils.getConfig(workflow.item, 'token');
+    if (avatarImg) updates.actor.img = avatarImg;
+    if (tokenImg) {
+        genericUtils.setProperty(updates, 'actor.prototypeToken.texture.src', tokenImg);
+        genericUtils.setProperty(updates, 'token.texture.src', tokenImg);
+    }
+    let hpFormula = 40 + (spellLevel - 4) * 10;
+    updates.actor.system.attributes.hp = {
+        formula: hpFormula,
+        max: hpFormula,
+        value: hpFormula
+    };
+    let animation = itemUtils.getConfig(workflow.item, 'animation') ?? 'none';
+    await Summons.spawn(sourceActor, updates, workflow.item, workflow.token, {
+        duration: 3600,
+        range: 90,
+        animation,
+        initiativeType: 'follows',
+        additionalSummonVaeButtons: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData].map(i => {return {type: 'use', name: i.name, identifier: i.flags['chris-premades'].info.identifier};})
+    });
+}
+async function attack({workflow}) {
+    if (workflow.targets.size !== 1 || workflow.advantage) return;
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'spiritOfDeathHauntCreature');
+    if (!effect) {
+        workflow.aborted = true;   
+        return;
+    }
+    let {targets: validTargetUuids, formula} = effect.flags['chris-premades'].spiritOfDeathHauntCreature;
+    if (!validTargetUuids.includes(workflow.targets.first().document.uuid)) {
+        workflow.aborted = true;
+        return;
+    }
+    workflow.advantage = true;
+    workflow.attackAdvAttribution.add(genericUtils.translate('DND5E.Advantage') + ': ' + effect.name);
+}
+async function late({workflow}) {
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'summonedEffect');
+    let originItem = fromUuidSync(effect?.origin);
+    let sourceActor = originItem?.actor;
+    if(!sourceActor)
+    {
+        return;
+    }
+    let featureData = await compendiumUtils.getItemFromCompendium(constants.packs.summonFeatures, 'Haunt Creature: Haunt', {object: true, getDescription: true, translate: 'CHRISPREMADES.Macros.SpiritOfDeath.HauntCreatureHaunt', identifier: 'spiritOfDeathHauntCreatureHaunt', flatDC: sourceActor.system.attributes.spelldc});
+    if (!featureData) {
+        errors.missingPackItem();
+        return;
+    }
+    let targetEffectData = {
+        name: genericUtils.translate('CHRISPREMADES.Macros.SpiritOfDeath.Haunted'),
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            dae: {
```

Comment:
I think I'd put `showIcon: true` in the dae flags, so it's evident who's haunted